### PR TITLE
feat: adds a default config for daily cluster for gpus

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
Creates a default configuration for GPU 

**Which issue(s) does this PR fix?**:

https://d2iq.atlassian.net/browse/D2IQ-93019
https://d2iq.atlassian.net/browse/D2IQ-92880

**Special notes for your reviewer**:

1. Create a GPU nodepool 
```
$ dkp create nodepool aws -c dkp-daily-3620240 --ami=ami-063f93c870491b17b --instance-type=p2.xlarge ubuntu20-p2
```
2. Enable the application 
3. apply the override cm 
```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: kommander
  name: nvidia-operator-overrides-cm
data:
  values.yaml: |
    nfd:
      enabled: false
    driver:
      enabled: false
    toolkit:
      version: v1.11.0-ubuntu20.04
    gfd:
      enabled: true
    dcgm:
      enabled: true
      version: 2.3.6-1-ubuntu20.04
    dcgmExporter:
      enabled: true
      version: 2.4.6-2.6.10-ubuntu20.04
    validator:
      image: mesosphere/gpu-operator-validator
      repository: docker.io
      version: v1.11.0
```
4. update Application to use this override 
```
apiVersion: apps.kommander.d2iq.io/v1alpha3
kind: AppDeployment
metadata:
  creationTimestamp: "2022-09-23T16:05:13Z"
  finalizers:
  - kommander.mesosphere.io/appsconfigfederation
  - kommander.mesosphere.io/appdeployment
  generation: 2
  name: nvidia-gpu-operator
  namespace: kommander
  resourceVersion: "1315299"
  uid: 2a44591c-5074-44c3-8349-6ebbe137f6ac
spec:
  appRef:
    kind: ClusterApp
    name: nvidia-gpu-operator-1.11.1
  configOverrides:
    name: nvidia-operator-overrides-cm
 ```

**Does this PR introduce a user-facing change?**:

```release-note

```

**If the PR adds a version bump, does it add a breaking change in License**:

- [ ] No License Change (or NA).

### CODE REVIEW

File: services/nvidia-gpu-operator/1.11.1/defaults/cm.yaml [hunk 1]
```diff
@@ -5,12 +5,22 @@ metadata:
...
     driver:
       enabled: false
     toolkit:
-      version: v1.10.0-centos7
+      version: v1.11.0-ubuntu20.04
     gfd:
       enabled: true
+    dcgm:
+      enabled: true
+      version: 2.3.6-1-ubuntu20.04
```
Comment (cbuto) [inline services/nvidia-gpu-operator/1.11.1/defaults/cm.yaml:19]: just out of curiosity, are these versions related to the underlying host OS? aka will they change depending on the OS being used in the cluster?

**Review by cbuto — APPROVED** [review]
lgtm!

I'm not sure if this is in scope, but insights flagged a bunch of [critical best practices](https://a5fdc7e85082e483b8247ab00ca9a104-1245960395.us-west-2.elb.amazonaws.com/dkp/kommander/dashboard/workspace/kommander-workspace/insights) for some of the nvidia pods. 

We might be able to address the findings specific to operator (run as non-root, disable privilege escalation) but we might not be able to do anything about the resources the operator creates.

File: services/nvidia-gpu-operator/1.11.1/defaults/cm.yaml (hunk 1 above)
Comment (faiq) [reply services/nvidia-gpu-operator/1.11.1/defaults/cm.yaml:26]: yes it will! thank you this comment. i added notes explaining why the values were chosen.

Comment (cbuto) [reply services/nvidia-gpu-operator/1.11.1/defaults/cm.yaml:26]: perfect, thanks!

File: services/nvidia-gpu-operator/1.11.1/grafana-dashboards/gpu-operator.json [hunk 2]
```diff
@@ -0,0 +1,985 @@
...
+      "5m",
+      "15m",
+      "30m",
+      "1h",
+      "2h",
+      "1d"
+    ]
+  },
+  "timezone": "",
+  "title": "NVIDIA DCGM Exporter Dashboard",
```
Comment (gracedo) [inline services/nvidia-gpu-operator/1.11.1/grafana-dashboards/gpu-operator.json:979]:
```suggestion
  "title": "Platform Apps / NVIDIA DCGM Exporter",
```

**Comment by gracedo** [comment]
@faiq Are there more steps to get this tested on the daily cluster? I ran the `dkp create nodepool` cmd but got stuck there. `dkp get nodepools` wasn't showing me the nodepool created in the create step 🤔

**Comment by faiq** [comment]
@gracedo it looks like you didnt set the clustername correctly 

```
07:10 PM  faiqraza @ linux  ~/go/src/github.com/mesosphere/ksphere-platform/cluster/konvoy2  master? 
$ kubectl get machinedeployments 
NAME                     CLUSTER             REPLICAS   READY   UPDATED   UNAVAILABLE   PHASE     AGE    VERSION
dkp-daily-3621941-md-0   dkp-daily-3621941   5          5       5         0             Running   18h    v1.24.4
ubuntu20-p2              dkp-daily-3620240                                                        110m   v1.23.7

07:10 PM  faiqraza @ linux  ~/go/src/github.com/mesosphere/ksphere-platform/cluster/konvoy2  master? 
$ dkp create nodepool aws -c dkp-daily-3621941 --ami=ami-063f93c870491b17b --instance-type=p2.xlarge ubuntu20-p2xlarge 
machinedeployment.cluster.x-k8s.io/ubuntu20-p2xlarge created
⠈⠁ Creating default/ubuntu20-p2xlarge nodepool resources awsmachinetemplate.infrastructure.cluster.x-k8s.io/ubuntu20-p2xlarge created
⠈⠑ Creating default/ubuntu20-p2xlarge nodepool resources kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/ubuntu20-p2xlarge created
 ✓ Creating default/ubuntu20-p2xlarge nodepool resources

07:10 PM  faiqraza @ linux  ~/go/src/github.com/mesosphere/ksphere-platform/cluster/konvoy2  master? 
$ kubectl get machinedeployments 
NAME                     CLUSTER             REPLICAS   READY   UPDATED   UNAVAILABLE   PHASE       AGE    VERSION
dkp-daily-3621941-md-0   dkp-daily-3621941   5          5       5         0             Running     18h    v1.24.4
ubuntu20-p2              dkp-daily-3620240                                                          111m   v1.23.7
ubuntu20-p2xlarge        dkp-daily-3621941   1                  1         1             ScalingUp   9s     v1.23.7
```

i just created a new one, but is on a slightly older k8s version. i think this should still be ok for testing the configuration

**Review by gracedo — APPROVED** [review]

**Comment by gracedo** [comment]
kommander tests https://github.com/mesosphere/kommander/pull/2382 are ✅  merging to get this into the daily build
